main.tf

```diff
@@ -1,116 +1,117 @@
+  role       = aws_iam_role.cloudwatch_role.id
+}
+
+data aws_iam_policy_document "assume_policy" {
+  statement {
+    actions = [
+      "sts:AssumeRole"
     ]
+    effect = "Allow"
+    principals {
+      type        = "Service"
+      identifiers = ["ec2.amazonaws.com"]
+    }
   }
 }
 
-resource "aws_cloudwatch_metric_alarm" "disk-full" {
-  alarm_name                = "${var.project_name}-${aws_instance.instance.id}-disk-full"
-  comparison_operator       = "GreaterThanOrEqualToThreshold"
-  evaluation_periods        = "3"
-  metric_name               = "DiskSpaceUtilization"
-  namespace                 = "System/Linux"
-  period                    = "60"
-  statistic                 = "Average"
-  threshold                 = "${var.disk_utilization_alarm_threshold}"
-  alarm_description         = "This metric monitors disk utilization"
-  alarm_actions = ["${lookup(var.bellosguardo_sns_topic_arn, var.bellosguardo_target)}"]
-  ok_actions = ["${lookup(var.bellosguardo_sns_topic_arn, var.bellosguardo_target)}"]
-  treat_missing_data = "breaching"
-  dimensions {
-    InstanceId = "${aws_instance.instance.id}"
-    MountPath = "/"
-    Filesystem = "overlay"
-  }
+resource "aws_iam_role" "cloudwatch_role" {
+  name                  = "${var.project_name}-ec2-cloudwatch-role"
+  force_detach_policies = true
+  assume_role_policy    = data.aws_iam_policy_document.assume_policy.json
 }
 
-variable "bellosguardo_sns_topic_arn" {
-  type = "map"
-  default = {
-    buildo = "arn:aws:sns:eu-west-1:309416224681:bellosguardo"
-    omnilab = "arn:aws:sns:eu-west-1:143727521720:bellosguardo"
-  }
+resource "aws_iam_instance_profile" "profile" {
+  name = "${var.project_name}-cloudwatch-profile"
+  role = aws_iam_role.cloudwatch_role.name
 }
 
-resource "aws_route53_record" "dns" {
-  count = "${length(var.zone_id) > 0 && length(var.host_name) > 0 ? 1 : 0}"
+resource "aws_cloudwatch_metric_alarm" "disk_full" {
+  count               = var.disk_utilization_alarm_threshold == 0 ? 0 : 1
+  alarm_name          = "${var.project_name}-${aws_instance.instance.id}-disk-full"
+  comparison_operator = "GreaterThanOrEqualToThreshold"
+  evaluation_periods  = "3"
+  metric_name         = "DiskSpaceUtilization"
+  namespace           = "System/Linux"
+  period              = "60"
+  statistic           = "Average"
+  threshold           = var.disk_utilization_alarm_threshold
+  alarm_description   = "This metric monitors disk utilization"
+  alarm_actions       = var.sns_alarm_enabled == true ? [var.sns_topic_alarm_arn] : []
+  ok_actions          = var.sns_alarm_enabled == true ? [var.sns_topic_alarm_arn] : []
+  treat_missing_data  = "breaching"
 
-  zone_id = "${var.zone_id}"
-  name = "${var.host_name}"
-  type = "A"
-  ttl = "300"
-  records = ["${aws_instance.instance.public_ip}"]
+  dimensions = {
+    InstanceId = aws_instance.instance.id
+    MountPath  = "/"
+    Filesystem = "overlay"
+  }
```

Comment:
Careful here, these dimensions do not match the ones created by the agent

Reply:
Now yes

Reply:
I beg to differ: you'll have to search for `InstanceId`, `path`, `device`, and `fstype`